explorer/bench/Bench/Pos/Explorer/ServerBench.hs

```diff
@@ -0,0 +1,215 @@
+    , _gtaSlotLeaders             :: {-Maybe -}SlotLeaders
+    } deriving (Generic)
+
+makeLenses ''GeneratedTestArguments
+
+instance HasConfigurations => NFData GeneratedTestArguments
+
+-- | More predictable generation that doesn't violate the invariants.
+genArgs :: Word -> Word -> Word -> IO GeneratedTestArguments
+genArgs blocksNumber slotsPerEpoch pageNumber = do
+
+    slotStart     <- liftIO $ generate $ arbitrary
+
+    blockHHs      <- liftIO $ generate $ arbitrary
+    blundsFHHs    <- liftIO $ generate $ withDefConfigurations arbitrary
+
+    slotLeaders   <- produceSlotLeaders blocksNumber
+    secretKeys    <- produceSecretKeys blocksNumber
+
+    blocks <- withDefConfigurations $
+        produceBlocksByBlockNumberAndSlots blocksNumber slotsPerEpoch slotLeaders secretKeys
+
+    let tipBlock = Prelude.last blocks
+
+    pure $ GeneratedTestArguments
+        { _gtaPageNumber              = Just pageNumber
+        , _gtaTipBlock                = tipBlock
+        , _gtaBlockHeaderHashes       = blockHHs
+        , _gtaBlundsFromHeaderHashes  = blundsFHHs
+        , _gtaSlotStart               = slotStart
+        , _gtaSlotLeaders             = slotLeaders
+        }
+
+----------------------------------------------------------------
+-- Mocked functions
+----------------------------------------------------------------
+
+getBlocksTotalMock
+    :: (HasConfigurations)
+    => GeneratedTestArguments
+    -> IO Integer
+getBlocksTotalMock genTestArgs = do
+    testParams <- testParamsGen
+
+    -- We replace the "real" database with our custom data.
+    let mode :: ExplorerMockMode
+        mode = def { emmGetTipBlock = pure $ genTestArgs ^. gtaTipBlock }
+
+    -- The extra context so we can mock the functions.
+    let extraContext :: ExtraContext
+        extraContext = makeMockExtraCtx mode
+
+    runExplorerTestMode testParams extraContext getBlocksTotal
+
+
+-- | The actual benched function.
+getBlocksPageMock
+    :: (HasConfigurations)
+    => GeneratedTestArguments
+    -> IO (Integer, [CBlockEntry])
+getBlocksPageMock genTestArgs = do
+    testParams <- testParamsGen -- TODO(ks): Temporary test params, will be removed.
+
+        -- We replace the "real" database with our custom data.
+    let mode :: ExplorerMockMode
+        mode = def {
+            emmGetTipBlock         = pure $ genTestArgs ^. gtaTipBlock,
+            emmGetPageBlocks       = \_ -> pure $ Just $ genTestArgs ^. gtaBlockHeaderHashes,
+            emmGetBlundFromHH      = \_ -> pure $ Just $ genTestArgs ^. gtaBlundsFromHeaderHashes,
+            emmGetSlotStart        = \_ -> pure $ Just $ genTestArgs ^. gtaSlotStart,
+            emmGetLeadersFromEpoch = \_ -> pure $ Just $ genTestArgs ^. gtaSlotLeaders
+        }
+
+    -- The extra context so we can mock the functions.
+    let extraContext :: ExtraContext
+        extraContext = makeMockExtraCtx mode
+
+    runExplorerTestMode testParams extraContext $ getBlocksPage pageNumber (Just 10)
```

Comment:
10 is the number of pages, it's what we use everywhere - when we modify that in other places, we can change it here, too.

Reply:
Constant of `pageSize` is defined here https://github.com/input-output-hk/cardano-sl/blob/master/explorer/src/Pos/Explorer/DB.hs#L94-L95

Reply:
Yes, it's the merge to master from *yesterday*.

Reply:
I'm not sure if we really have to put generated file `ServerBackend.html` into repository. It will be changed on my machine if I run `stack bench cardano-sl-explorer`. I guess this will be happened anytime if anyone run benchmarks on its machine.

Reply:
Notice the comment here - https://github.com/input-output-hk/cardano-sl/pull/1526#issuecomment-339933853